### ISSUE TITLE
 which api trojan-panel-core calls i need to recode it

### ISSUE THREAD

**Issue by netsupcloud** [trojanpanel/install-script#352]
which api trojan-panel-core calls i need to recode it     just give me logic and interfaces that core with call them

i will rewrite it for myself

**Comment by jonssonyan** [comment]
This is what trojan-panel-core does. Its function is to call the api interface of xray or hysteria, such as querying users, querying quotas, and storing the results in MariaDB and Redis.

**Comment by netsupcloud** [comment]
> This is what trojan-panel-core does. Its function is to call the api interface of xray or hysteria, such as querying users, querying quotas, and storing the results in MariaDB and Redis.

ok i think its easy in container run a script.sh that will run single trojan-panel-core and will handle its exit code and will restart it instead restart all of container

and another script will handle state of xray cores and will restart them if was need instead restarting all of container

**Comment by jonssonyan** [comment]
xray/hysteria/naiveproxy runs in the trojan-panel-core container through coroutines. Restarting the container will restart all programs in the container. And for hysteria/hysteria2, it is authenticated through HTTP authentication. If there is no trojan-panel -core, they will not make sense.

**Comment by netsupcloud** [comment]
yes it right but its not a good idea when trojan-panel-core can be under pressure when some one will create 200 nodes
best schematic:
                                         ------------tp-core-runner.py.js.sh-------trojan-panel-core
tp-core-container---------
                                         ------------core-handler.py.js.sh-----------(xray)(xray)(xray)(xray)(xray)(xray)(xray)(hysteria)(trojan-go)


and trojan-panel-core must use coroutines and call to core-handler.py.js.sh instead directly create nodes

**Comment by netsupcloud** [comment]
and of course you must handle recovery of trojan-panel-core after crash and getting back it to working

**Comment by netsupcloud** [comment]
or if you want avoid using (core-handler.py.js.sh.go) you must find a way to run xray cores as non-children of trojan-panel-core

and anyway you must use tp-core-runner.py.js.sh.go for both type of structure
